src/Data/Boring.hs

```diff
@@ -186,6 +190,24 @@ instance (Boring a, Boring b, Boring c, Boring d, Boring e) => Boring (a, b, c,
 instance Boring a => Boring (Stream a) where
     boring = boring :> boring
 
+class GBoring f where
```

Comment:
I wonder if we want to export this class, and maybe also define a helper `genericallyBoring = G.to gboring`

-- and the same for `Absurd`